File: Solutions/0076/0076.go
func minWindow(s string, t string) string {
	count := make([]int, 128)
	required := len(t)
	bestLeft := -1
	minLength := len(s) + 1

	for _, c := range t {
		count[c]++
	}

	for l, r := 0, 0; r < len(s); r++ {
		if count[s[r]]--; count[s[r]] >= 0 {
			required--
		}
		for required == 0 {
			if r-l+1 < minLength {
				bestLeft = l
				minLength = r - l + 1
			}
			if count[s[l]]++; count[s[l]] > 0 {
				required++
			}
			l++
		}
	}

	if bestLeft == -1 {
		return ""
	}

	return s[bestLeft : bestLeft+minLength]
}

File: Solutions/0076/0076.js
var minWindow = function(s, t) {
    const count = new Array(128).fill(0);
  let required = t.length;
  let bestLeft = -1;
  let minLength = s.length + 1;

  for (const c of t) {
    ++count[c.charCodeAt(0)];
  }

  for (let l = 0, r = 0; r < s.length; ++r) {
    if (--count[s.charCodeAt(r)] >= 0) {
      --required;
    }
    while (required === 0) {
      if (r - l + 1 < minLength) {
        bestLeft = l;
        minLength = r - l + 1;
      }
      if (++count[s.charCodeAt(l++)] > 0) {
        ++required;
      }
    }
  }

  return bestLeft === -1 ? "" : s.substring(bestLeft, bestLeft + minLength);
};